target/linux/ramips/dts/GB-PC1.dts

```diff
@@ -0,0 +1,157 @@
+/dts-v1/;
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "GB-PC1";
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x1c000000>, <0x20000000 0x4000000>;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,57600";
+	};
+
+	palmbus: palmbus@1E000000 {
+		i2c@900 {
+			status = "okay";
+		};
+	};
+
+	gpio-keys-polled {
+		compatible = "gpio-keys-polled";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio0 18 GPIO_ACTIVE_HIGH>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+
+		system {
+			label = "gb-pc1:green:system";
+			gpios = <&gpio0 6 GPIO_ACTIVE_LOW>;
+		};
+
+		status {
+			label = "gb-pc1:green:status";
+			gpios = <&gpio0 8 GPIO_ACTIVE_LOW>;
+		};
+
+		lan {
+			label = "gb-pc1:green:lan";
+			gpios = <&gpio0 24 GPIO_ACTIVE_LOW>;
+		};
+
+		wan {
+			label = "gb-pc1:green:wan";
+			gpios = <&gpio0 25 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	gpio-export {
+		compatible = "gpio-export";
+		#size-cells = <0>;
+
+		gpio26 {
+			gpio-export,name = "gpio26";
+			gpio-export,direction_may_change = <1>;
+			gpios = <&gpio0 26 GPIO_ACTIVE_HIGH>;
+		};
+
+		gpio28 {
+			gpio-export,name = "gpio28";
+			gpio-export,direction_may_change = <1>;
+			gpios = <&gpio0 28 GPIO_ACTIVE_HIGH>;
+		};
+
+		gpio29 {
+			gpio-export,name = "gpio29";
+			gpio-export,direction_may_change = <1>;
+			gpios = <&gpio0 29 GPIO_ACTIVE_HIGH>;
+		};
+
+		gpio30 {
+			gpio-export,name = "gpio30";
+			gpio-export,direction_may_change = <1>;
+			gpios = <&gpio0 30 GPIO_ACTIVE_HIGH>;
+		};
+	};
+};
+
+&sdhci {
+	status = "okay";
+};
+
+&spi0 {
+	status = "okay";
+
+	m25p80@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+		m25p,chunked-io = <32>;
+
+		partition@0 {
+			label = "u-boot";
+			reg = <0x0 0x30000>;
+			read-only;
+		};
+
+		partition@30000 {
+			label = "u-boot-env";
+			reg = <0x30000 0x10000>;
+		};
+
+		factory: partition@40000 {
+			label = "factory";
+			reg = <0x40000 0x10000>;
+			read-only;
+		};
+
+		partition@50000 {
+			label = "firmware";
+			reg = <0x50000 0x1fb0000>;
+		};
+	};
+};
+
+&cpuclock {
```

Comment:
The .dts does describe the hardware...but not the Mediatek default clock setting meant to insure functionality with DDR2 RAM and other design factors.
I spent many hours and much money during the design and testing of this hardware.
That includes days of running the prototypes and other MT7621 devices at a higher clock rate and running tests such as 'stress'.
The ODM/OEM (me) shipped the devices using these clock settings.
Do all MT7621 devices from the "factory" have the clock set @ 880MHz?
And if so why can I not set my clock @ 2GHz ?